### PR TITLE
[DO NOT EVEN THINKY ABOUT MERGY]

### PR DESCRIPTION
**Pull request by dbouwman**

### CODE REVIEW

File: addon/mixins/service-mixin.js [hunk 1]
```diff
@@ -176,20 +176,38 @@ export default Mixin.create({
...
+    // ---------------------------------------------------
     // exactly the same token logic as in requestUrl() above
-    const token = portalOpts ? portalOpts.token : this.get('session.token');
-    if (token) {
-      if (!args.params) {
-        args.params = {};
+    if (portalOpts && portalOpts.token) {
+      // for some reason we REALLY want to use this token...
+      // so we construct a minimalist IAuthenticationManger
+      let portalToUse = portalOpts.portalHostname || this.get('session.portalHostname');
```
Comment (tomwayson) [inline addon/mixins/service-mixin.js]: These 3 lines should be replaced w/ `let portalToUse = this.getPortalRestUrl(portalOpts);` Let it do it's thing (which includes [fixing the url for enterprise environments](https://github.com/Esri/ember-arcgis-portal-services/blob/v1.5.0/addon/mixins/service-mixin.js#L54-L58)).

File: addon/mixins/service-mixin.js [hunk 2]
```diff
@@ -176,20 +176,38 @@ export default Mixin.create({
...
+      this.set('session.authMgr.portal', correctPortalRestUrl);
+    }
+    // ---------------------------------------------------
     // exactly the same token logic as in requestUrl() above
-    const token = portalOpts ? portalOpts.token : this.get('session.token');
-    if (token) {
-      if (!args.params) {
-        args.params = {};
+    if (portalOpts && portalOpts.token) {
+      // for some reason we REALLY want to use this token...
```
Comment (tomwayson) [inline addon/mixins/service-mixin.js]: // user wants to override the portal and/or token for this request

File: addon/mixins/service-mixin.js [hunk 3]
```diff
@@ -176,20 +176,38 @@ export default Mixin.create({
...
+        portal: portalToUse,
+        getToken: function () {
+          return Promise.resolve(portalOpts.token);
+        }
+      };
+    } else {
+      // use the session authMgr
+      args.authentication = authMgr;
     }
+    // debug(`addOptions is returning ${JSON.stringify(args, null, 4)}`);
```
Comment (tomwayson) [inline addon/mixins/service-mixin.js]: remove comment

**Comment by dbouwman** [comment]
#133 supersedes this PR. closing

**Pull request merged**